### PR TITLE
Add some integration tests for packages

### PR DESCRIPTION
**Pull request by stevededalus**

### CODE REVIEW

File: api/python/tests/integration/test_packages.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+def mock_make_api_call(operation_name):
+    """ Mock boto3's AWS API Calls for testing. """
+    if operation_name == 'GetObject':
+        parsed_response = {'Body': {'foo'}}
+        return parsed_response
+    raise NotImplementedError
+
+def test_read_manifest(tmpdir):
+    """ Verify reading serialized manifest from disk. """
+    pkg = Package.load(open(LOCAL_MANIFEST))
```
Comment (dimaryaz) [inline api/python/tests/integration/test_packages.py]: Use `with open(...) as fd:`

File: api/python/tests/integration/test_packages.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+
+    with pytest.raises(NotImplementedError):
+        pkg.get('foo')
+
+    out_path = os.path.join(tmpdir, 'new_manifest.jsonl')
+    pkg.dump(open(out_path, "w"))
```
Comment (dimaryaz) [inline api/python/tests/integration/test_packages.py]: Same here. Here it actually matters cause the data might not get written until the file object is garbage collected at some arbitrary time in the future.

File: api/python/tests/integration/test_packages.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    
+    # Insepct the jsonl to verify everything is maintained, i.e.
+    # that load/dump results in an equivalent set.
+    original_set = list(jsonlines.Reader(open(LOCAL_MANIFEST)))
```
Comment (dimaryaz) [inline api/python/tests/integration/test_packages.py]: Wouldn't it be easier to just `Package.load` the second file and compare the contents of the packages? That would verify that write/read works without depending on the internals of it.

File: api/python/tests/integration/test_packages.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    written_set = list(jsonlines.Reader(open(out_path)))
+    assert len(original_set) == len(written_set)
+    assert sorted(original_set, key=lambda k: k.get('logical_key','manifest')) \
+        == sorted(written_set, key=lambda k: k.get('logical_key','manifest'))
+
+def test_materialize_from_remote():
+    """ Verify loading data and mainfest transforms from S3. """
+    with patch('botocore.client.BaseClient._make_api_call', new=mock_make_api_call):
```
Comment (dimaryaz) [inline api/python/tests/integration/test_packages.py]: It's fine for something simple like this... But in general, look into `botocore.stub.Stubber`. See the old registry tests for examples: https://github.com/quiltdata/quilt/blob/master/registry/tests/utils.py

File: api/python/tests/integration/test_packages.py (hunk 3 above)
Comment (stevededalus) [reply api/python/tests/integration/test_packages.py]: Added a todo for now, needs the __eq__ to make this easier to work with.

File: api/python/tests/integration/test_packages.py (hunk 4 above)
Comment (stevededalus) [reply api/python/tests/integration/test_packages.py]: Will take a look at that in a follow up.

File: api/python/tests/integration/test_packages.py (hunk 3 above)
Comment (dimaryaz) [reply api/python/tests/integration/test_packages.py]: Ok. Can you use `with open` here and below, though?

**Pull request merged**